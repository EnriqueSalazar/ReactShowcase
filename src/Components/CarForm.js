/**
 * Created by enriq on 17/11/16.
 */
import React from "react";
import FormItem from "../Components/FormItem";
import { Grid, Col, Row, Well, Button, Glyphicon } from "react-bootstrap";

const CarForm = props => {
  let {
    filterBrands,
    filterBrandColors,
    selectedColor,
    vehicleTypes,
    vehicleBrands,
    vehicleBrandColors,
    addSelected
  } = props;
  return (
    <div>
      <Well>
        <Grid fluid>
          <Row>
            <Col md={3}>
              <FormItem
                callback={filterBrands}
                label="Type"
                list={vehicleTypes}
              />
            </Col>
            <Col md={3}>
              <FormItem
                callback={filterBrandColors}
                label="Brands"
                list={vehicleBrands}
              />
            </Col>
            <Col md={3}>
              <FormItem
                callback={selectedColor}
                label="Colors"
                list={vehicleBrandColors}
              />
            </Col>
            <Col md={3}>
              <br />
              <Button
                bsStyle="primary"
                bsSize="large"
                onClick={() => addSelected()}
                block
              >
                <Glyphicon glyph="ok" />{" Select"}
              </Button>
            </Col>
          </Row>
        </Grid>
      </Well>
    </div>
  );
};

export default CarForm;
