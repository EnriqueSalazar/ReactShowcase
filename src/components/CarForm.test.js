/**
 * Created by enriq on 17/11/16.
 */
import React from "react";
import CarForm from ".//CarForm";
import renderer from "react-test-renderer";

test("Render CarForm test", () => {
  let testCallback = () => {};
  const component = renderer.create(
    (
      <CarForm
        filterBrands={testCallback}
        vehicleTypes={
          [ { value: 1, option: "uno" }, { value: 2, option: "dos" } ]
        }
        filterBrandColors={testCallback}
        vehicleBrands={
          [ { value: 3, option: "tres" }, { value: 4, option: "cuatro" } ]
        }
        selectedColor={testCallback}
        vehicleBrandColors={
          [ { value: 5, option: "cinco" }, { value: 6, option: "seis" } ]
        }
        addSelected={testCallback}
      />
    )
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
