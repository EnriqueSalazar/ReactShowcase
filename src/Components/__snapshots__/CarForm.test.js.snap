exports[`test Render CarForm test 1`] = `
<div>
  <div
    className="well">
    <div
      className="container-fluid">
      <div
        className="row">
        <div
          className="col-md-3">
          <div
            className="form-group">
            <label
              className="control-label"
              htmlFor="formControlsSelect">
              Type
            </label>
            <select
              className="form-control"
              id="formControlsSelect"
              onChange={[Function]}
              type={undefined}>
              <option
                value="select">
                
              </option>
              <option
                value={1}>
                uno
              </option>
              <option
                value={2}>
                dos
              </option>
            </select>
          </div>
        </div>
        <div
          className="col-md-3">
          <div
            className="form-group">
            <label
              className="control-label"
              htmlFor="formControlsSelect">
              Brands
            </label>
            <select
              className="form-control"
              id="formControlsSelect"
              onChange={[Function]}
              type={undefined}>
              <option
                value="select">
                
              </option>
              <option
                value={3}>
                tres
              </option>
              <option
                value={4}>
                cuatro
              </option>
            </select>
          </div>
        </div>
        <div
          className="col-md-3">
          <div
            className="form-group">
            <label
              className="control-label"
              htmlFor="formControlsSelect">
              Colors
            </label>
            <select
              className="form-control"
              id="formControlsSelect"
              onChange={[Function]}
              type={undefined}>
              <option
                value="select">
                
              </option>
              <option
                value={5}>
                cinco
              </option>
              <option
                value={6}>
                seis
              </option>
            </select>
          </div>
        </div>
        <div
          className="col-md-3">
          <br />
          <button
            className="btn btn-lg btn-primary btn-block"
            disabled={false}
            onClick={[Function]}
            type="button">
            <span
              className="glyphicon glyphicon-ok" />
             Select
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
